extensions/vault/runtime/src/main/java/io/quarkus/vault/runtime/client/MutinyVertxVaultClient.java

```diff
@@ -268,129 +266,133 @@ public VaultWrapResult wrap(String token, long ttl, Object object) {
-        Request request = builder(path, token).get().build();
+        HttpRequest<Buffer> request = builder(path, token).method(HttpMethod.GET);
         return exec(request, resultClass);
     }
 
     protected <T> T get(String path, Map<String, String> queryParams, Class<T> resultClass) {
-        final Request request = builder(path, queryParams).get().build();
+        final HttpRequest<Buffer> request = builder(path, queryParams).method(HttpMethod.GET);
         return exec(request, resultClass);
     }
 
     protected int head(String path) {
-        final Request request = builder(path).head().build();
+        final HttpRequest<Buffer> request = builder(path).method(HttpMethod.HEAD);
         return exec(request);
     }
 
     protected int head(String path, Map<String, String> queryParams) {
-        final Request request = builder(path, queryParams).head().build();
+        final HttpRequest<Buffer> request = builder(path, queryParams).method(HttpMethod.HEAD);
         return exec(request);
     }
 
-    private <T> T exec(Request request, Class<T> resultClass) {
-        return exec(request, resultClass, 200);
+    private <T> T exec(HttpRequest<Buffer> request, Class<T> resultClass) {
+        return exec(request, null, resultClass, 200);
+    }
+
+    private <T> T exec(HttpRequest<Buffer> request, int expectedCode) {
+        return exec(request, null, null, expectedCode);
     }
 
-    private <T> T exec(Request request, int expectedCode) {
-        return exec(request, null, expectedCode);
+    private <T> T exec(HttpRequest<Buffer> request, Object body, Class<T> resultClass) {
+        return exec(request, body, resultClass, 200);
     }
 
-    private <T> T exec(Request request, Class<T> resultClass, int expectedCode) {
-        try (Response response = client.newCall(request).execute()) {
-            if (response.code() != expectedCode) {
+    private <T> T exec(HttpRequest<Buffer> request, Object body, Class<T> resultClass, int expectedCode) {
+        try {
+            HttpResponse<Buffer> response = body == null
+                    ? request.sendAndAwait()
+                    : request.sendJsonAndAwait(requestBody(body));
+            if (response.statusCode() != expectedCode) {
                 throwVaultException(response);
             }
-            String jsonBody = response.body().string();
+            String jsonBody = response.body().toString();
             return resultClass == null ? null : mapper.readValue(jsonBody, resultClass);
-        } catch (IOException e) {
+        } catch (Exception e) {
             throw new VaultException(e);
         }
     }
 
-    private int exec(Request request) {
-        try (Response response = client.newCall(request).execute()) {
-            return response.code();
-        } catch (IOException e) {
+    private int exec(HttpRequest<Buffer> request) {
+        try {
+            return request.sendAndAwait().statusCode();
+        } catch (Exception e) {
```

Comment:
I had decided to only catch expected checked exceptions, so namely `IOException`.
did you decide to capture everything on purpose?

Reply:
So, await() will throw CompletionException or RuntimeException (wrapping the cause). So it's always RuntimeException.

Reply:
Ultimately, once we return a `Uni`, we would not have to do this.